diff --git a/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java b/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
index 2499c74165a4a7340220d64fed49cfbf8065777c..5a8a5d219d26a6be6f703966155ebb7f392a42bd 100644
--- a/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
+++ b/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
@@ -181,7 +181,13 @@ public class CameraRollModule extends NativeCameraRollModuleSpec {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
           ContentValues mediaDetails = new ContentValues();
           if (isAlbumPresent) {
-            String relativePath = Environment.DIRECTORY_DCIM + File.separator + album;
+            final String environment;
+            if ("video".equals(mOptions.getString("type"))) {
+              environment = Environment.DIRECTORY_MOVIES;
+            } else {
+              environment = Environment.DIRECTORY_PICTURES;
+            }
+            String relativePath = environment + File.separator + album;
             mediaDetails.put(MediaStore.MediaColumns.RELATIVE_PATH, relativePath);
           }
           mediaDetails.put(MediaStore.MediaColumns.MIME_TYPE, mimeType);
