diff --git a/node_modules/react-native-blob-util/ios/ReactNativeBlobUtilFS.h b/node_modules/react-native-blob-util/ios/ReactNativeBlobUtilFS.h
index f5267e03..dd54bc64 100644
--- a/node_modules/react-native-blob-util/ios/ReactNativeBlobUtilFS.h
+++ b/node_modules/react-native-blob-util/ios/ReactNativeBlobUtilFS.h
@@ -18,7 +18,15 @@
 #import "RCTBridgeModule.h"
 #endif
 
+#if __has_include(<AssetsLibrary/AssetsLibrary.h>) && (__IPHONE_OS_VERSION_MAX_ALLOWED < 180000)
 #import <AssetsLibrary/AssetsLibrary.h>
+#define RCTBLOBUTIL_HAS_ASSETSLIB 1
+#else
+#define RCTBLOBUTIL_HAS_ASSETSLIB 0
+@class ALAssetRepresentation;
+@class ALAsset;
+@class ALAssetsLibrary;
+#endif
 
 @interface ReactNativeBlobUtilFS : NSObject <NSStreamDelegate>  {
     NSOutputStream * outStream;
diff --git a/node_modules/react-native-blob-util/ios/ReactNativeBlobUtilFS.mm b/node_modules/react-native-blob-util/ios/ReactNativeBlobUtilFS.mm
index b7f59726..0a724778 100644
--- a/node_modules/react-native-blob-util/ios/ReactNativeBlobUtilFS.mm
+++ b/node_modules/react-native-blob-util/ios/ReactNativeBlobUtilFS.mm
@@ -11,7 +11,9 @@
 #import "ReactNativeBlobUtilFS.h"
 #import "ReactNativeBlobUtilConst.h"
 #import "ReactNativeBlobUtilFileTransformer.h"
+#if RCTBLOBUTIL_HAS_ASSETSLIB
 #import <AssetsLibrary/AssetsLibrary.h>
+#endif
 
 #import <CommonCrypto/CommonDigest.h>
 
@@ -186,6 +188,7 @@ + (void) readStream:(NSString *)uri
                 }
                 [stream close];
             }
+#if RCTBLOBUTIL_HAS_ASSETSLIB
             else if (asset != nil)
             {
                 int cursor = 0;
@@ -200,6 +203,7 @@ + (void) readStream:(NSString *)uri
                     }
                 }
             }
+#endif
             else
             {
                 NSDictionary * payload = @{ @"streamId":streamId, @"event": FS_EVENT_ERROR, @"code": @"EINVAL", @"detail": @"Unable to resolve URI" };
@@ -311,6 +315,7 @@ + (NSNumber *) writeFileFromFile:(NSString *)src toFile:(NSString *)dest append:
             __block NSNumber * size = [NSNumber numberWithLong:written];
             callback(nil, size);
         }
+#if RCTBLOBUTIL_HAS_ASSETSLIB
         else if(asset != nil)
         {
 
@@ -329,6 +334,7 @@ + (NSNumber *) writeFileFromFile:(NSString *)src toFile:(NSString *)dest append:
             [os close];
             callback(nil, size);
         }
+#endif
         else
             callback(@"failed to resolve path", nil);
     }];
@@ -500,6 +506,7 @@ + (void) readFile:(NSString *)path
         __block NSData * fileContent;
         NSError * err;
         __block Byte * buffer;
+#if RCTBLOBUTIL_HAS_ASSETSLIB
         if(asset != nil)
         {
             int size = asset.size;
@@ -515,6 +522,7 @@ + (void) readFile:(NSString *)path
             free(buffer);
         }
         else
+#endif
         {
             BOOL isDir = NO;
             if(![[NSFileManager defaultManager] fileExistsAtPath:path isDirectory: &isDir]) {
@@ -803,10 +811,12 @@ + (void) exists:(NSString *) path callback:(RCTResponseSenderBlock)callback
             exists = [[NSFileManager defaultManager] fileExistsAtPath:path isDirectory: &isDir];
             callback(@[@(exists), @(isDir)]);
         }
+#if RCTBLOBUTIL_HAS_ASSETSLIB
         else if(asset != nil)
         {
             callback(@[@YES, @NO]);
         }
+#endif
         else
         {
             callback(@[@NO, @NO]);
@@ -943,6 +953,7 @@ + (void)slice:(NSString *)path
             [os close];
             resolve(dest);
         }
+#if RCTBLOBUTIL_HAS_ASSETSLIB
         else if (asset != nil)
         {
             long expected = [end longValue] - [start longValue];
@@ -979,6 +990,7 @@ + (void)slice:(NSString *)path
             [os close];
             resolve(dest);
         }
+#endif
         else {
             reject(@"EINVAL", [NSString stringWithFormat: @"Could not resolve URI %@", path ], nil);
         }
@@ -1005,6 +1017,7 @@ + (void) getPathFromUri:(NSString *)uri completionHandler:(void(^)(NSString * pa
 {
     if([uri hasPrefix:AL_PREFIX])
     {
+#if RCTBLOBUTIL_HAS_ASSETSLIB
         NSURL *asseturl = [NSURL URLWithString:uri];
         __block ALAssetsLibrary* assetslibrary = [[ALAssetsLibrary alloc] init];
         [assetslibrary assetForURL:asseturl
@@ -1015,6 +1028,9 @@ + (void) getPathFromUri:(NSString *)uri completionHandler:(void(^)(NSString * pa
                       failureBlock:^(NSError *error) {
                           onComplete(nil, nil);
                       }];
+#else
+        onComplete(nil, nil);
+#endif
     }
     else
     {
@@ -1046,6 +1062,9 @@ +(void) df:(RCTResponseSenderBlock)callback
 
 + (void) writeAssetToPath:(ALAssetRepresentation * )rep dest:(NSString *)dest
 {
+#if !RCTBLOBUTIL_HAS_ASSETSLIB
+    return;
+#else
     int read = 0;
     int cursor = 0;
     Byte * buffer = (Byte *)malloc(10240);
@@ -1059,6 +1078,7 @@ + (void) writeAssetToPath:(ALAssetRepresentation * )rep dest:(NSString *)dest
     [ostream close];
     free(buffer);
     return;
+#endif
 }
 
 @end
diff --git a/node_modules/react-native-blob-util/ios/ReactNativeBlobUtil/ReactNativeBlobUtil.mm b/node_modules/react-native-blob-util/ios/ReactNativeBlobUtil/ReactNativeBlobUtil.mm
index 60f006a0..c48463ea 100644
--- a/node_modules/react-native-blob-util/ios/ReactNativeBlobUtil/ReactNativeBlobUtil.mm
+++ b/node_modules/react-native-blob-util/ios/ReactNativeBlobUtil/ReactNativeBlobUtil.mm
@@ -536,10 +536,14 @@ - (void)ls:(NSString *)path
         }
         else if(asset != nil)
         {
+#if RCTBLOBUTIL_HAS_ASSETSLIB
             __block NSNumber * size = [NSNumber numberWithLong:[asset size]];
             result = [asset metadata];
             [result setValue:size forKey:@"size"];
             callback(@[[NSNull null], result]);
+#else
+            callback(@[@"failed to stat path, assets-library is unsupported", [NSNull null]]);
+#endif
         }
         else
         {
@@ -601,8 +605,12 @@ - (void)cp:(NSString *)src
         NSError * error = nil;
         if(path == nil)
         {
+#if RCTBLOBUTIL_HAS_ASSETSLIB
             [ReactNativeBlobUtilFS writeAssetToPath:asset dest:dest];
             callback(@[[NSNull null], @YES]);
+#else
+            callback(@[@"failed to copy path, assets-library is unsupported", @NO]);
+#endif
         }
         else
         {
diff --git a/node_modules/react-native-blob-util/react-native-blob-util.podspec b/node_modules/react-native-blob-util/react-native-blob-util.podspec
index 98040b9f..f46c9397 100644
--- a/node_modules/react-native-blob-util/react-native-blob-util.podspec
+++ b/node_modules/react-native-blob-util/react-native-blob-util.podspec
@@ -17,7 +17,6 @@ Pod::Spec.new do |s|
     'ReactNativeBlobUtilPrivacyInfo' => ['ios/PrivacyInfo.xcprivacy'],
   }
   s.platforms       = { :ios => "11.0" }
-  s.framework    = 'AssetsLibrary'
 
   if fabric_enabled
     # Use install_modules_dependencies helper to install the dependencies if React Native version >=0.71.0.
